torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,429 @@
+            rename_dest = os.path.join(
+                dest_split[0],
+                dest_split[1].split("_")[0] + "_" + dest_split[1].split("_")[1],
+            )
+        if "_" not in modality:
+            rename_dest = os.path.join(dest_split[0], dest_split[1].split("_")[0])
+        os.rename(destination_folder, rename_dest)
+
+    def _convert_to_color(
+        self, arr_2d: Tensor, cmap: dict[int, tuple[int, int, int]]
+    ) -> "np.typing.NDArray[np.uint8]":
+        """Numeric labels to RGB-color encoding.
+
+        Args:
+            arr_2d: 2D array to be colorized
+            cmap: colormap to use when mapping the labels
+
+        Returns:
+            3D colored image
+        """
+        arr_3d = np.zeros((arr_2d.shape[0], arr_2d.shape[1], 3), dtype=np.uint8)
+
+        for c, i in cmap.items():
+            m = arr_2d == c
+            arr_3d[m] = i
+        return arr_3d
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample image-mask pair returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        modality_channels = {"viirs": 1, "esa_wc": 1, "s1": 2}
+
+        # Assign a number of channels to all 's2' modalities
+        for modality in self.modality:
+            if modality.startswith("s2"):
+                modality_channels[modality] = 10
+
+        start_idx = 0
+        split_images = {}
+
+        for modality in self.modality:
+            end_idx = start_idx + modality_channels[modality]  # Start + n of channels
+            split_images[modality] = sample["image"][start_idx:end_idx, :, :]  # Slicing
+            start_idx = end_idx  # Update the iterator
+
+        # Prepare the mask
+        mask = sample["mask"].squeeze()
+        color_mask = self._convert_to_color(mask, cmap=self.mask_cmap)
+
+        num_subplots = len(split_images) + 1  # +1 for color_mask
+        showing_predictions = "prediction" in sample
+        if showing_predictions:
+            num_subplots += 1
+
+        fig, axs = plt.subplots(1, num_subplots, figsize=(num_subplots * 4, 5))
+
+        # Plot each modality in its respective axis
+        for i, (modality, image) in enumerate(split_images.items()):
+            ax = axs[i]
+            img = np.transpose(image, (1, 2, 0)).squeeze()
+            # Apply transformations based on modality type
+            if modality.startswith("s2"):
+                img = img[:, :, [4, 3, 2]]
+            if modality == "esa_wc":
+                img = self._convert_to_color(torch.as_tensor(img), cmap=self.wc_cmap)
+            if modality == "s1":
```

Comment:
Is there a go-to viz scheme for Sentinel-1? I simply use the VV band here but can do a color composite as well.

Reply:
I like color composites (see the Sentinel1 dataset) but had trouble getting them to work for other scales (see the SSL4EOS12 dataset). I think either is fine.

Reply:
Alright will keep it this way for now then. Will be doing side experiments for this on the side.